src/VecSim/vec_sim_tiered_index.h

```diff
@@ -0,0 +1,41 @@
+#pragma once
+
+#include "vec_sim_index.h"
+#include "algorithms/brute_force/brute_force.h"
+
+#include <shared_mutex>
+
+template <typename DataType, typename DistType>
+class VecSimTieredIndex : public VecSimIndexInterface {
+protected:
+    BruteForceIndex<DataType, DistType> *tempFlat;
+    VecSimIndexAbstract<DistType> *index;
+
+    void *jobQueue;
+    SubmitCB SubmitJobsToQueue;
+
+    void *memoryCtx;
+    UpdateMemoryCB UpdateIndexMemory;
```

Comment:
Consider documenting what are the expectations from these callbacks and pointers (here or with the type definitions)

Reply:
I added docus in `vec_sim_common.h`